server_setup.coffee

```diff
@@ -122,7 +127,7 @@ setupCountryRedirectMiddleware = (app, country="china", countryCode="CN", langua
 
   app.use (req, res, next) ->
     if shouldRedirectToCountryServer req
-      res.writeHead 302, "Location": config[country + 'Domain'] + req.url
+      res.writeHead 302, "Location": "http://#{host} + '/' + #{req.url}"
```

Comment:
`"http://#{host}/#{req.url}"`– otherwise the plus, spaces, and single quotes are in your string still. (Was it working somehow, or haven't tested yet?)